compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEOps.td

```diff
@@ -556,6 +587,151 @@ def AMDAIE_NpuDmaCpyNdOp: AMDAIE_Op<"npu.dma_cpy_nd", [
+    whether a packet header is needed.
+
+    The representation supports a partially-static representation for the 
+    `offsets`, `sizes` and `strides`. A special sentinel value 
+    ShapedType::kDynamic encodes that the corresponding entry has a dynamic
+    value.
+
+    Example:
+
+    ```mlir
+    %2 = amdaie.connection(%1, %0) 
+      : (!amdaie.logicalobjectfifo<memref<32x64xi32, 1>>,
+      !amdaie.logicalobjectfifo<memref<32x1024xi32>>)
+    %bd_id = amdaie.bd_id(%tile_0_0, 0)
+    %channel = amdaie.channel(%tile_0_0, 0, port_type = DMA, direction = MM2S)
+    ...
+    amdaie.controlcode {
+      %5 = amdaie.logicalobjectfifo.from_memref %0, {%tile_0_0} 
+        : memref<32x1024xi32> -> !amdaie.logicalobjectfifo<memref<32768xi32>>
+      %4 = amdaie.npu.half_dma_cpy_nd async %2(%0[0, 0] [32, 64] [1024, 1]
+        bd_id = %bd_id channel = %channel)
+      ...
+    }
+    ```
+  }];
+
+  let arguments = (
+    ins Index:$connection,
+        AnyAMDAIELogicalObjectFifoType:$input,
+        Variadic<Index>:$offsets,
+        Variadic<Index>:$sizes,
+        Variadic<Index>:$strides,
+        DenseI64ArrayAttr:$static_offsets,
+        DenseI64ArrayAttr:$static_sizes,
+        DenseI64ArrayAttr:$static_strides,
+        Optional<Index>:$bd_id,
+        Optional<Index>:$channel
+  );
+
+  let results = (outs Optional<AMDAIE_AsyncTokenType>:$async_token);
+
+  let assemblyFormat = [{
+    custom<AsyncTokenType>(type($async_token))
+    $connection
+    `(`
+    $input
+    custom<DynamicIndexList>($offsets, $static_offsets)
+    custom<DynamicIndexList>($sizes, $static_sizes)
+    custom<DynamicIndexList>($strides, $static_strides)
+    (`bd_id` `=` $bd_id^)?
+    (`channel` `=` $channel^)?
+    `)`
+    attr-dict 
+    `:` type($input)
+  }];
+
+  let builders = [
+    // Build a NpuHalfDmaCpyNdOp with mixed static and dynamic entries.
+    OpBuilder<(ins "::mlir::TypeRange":$result_types, "Value":$connection,
+      "::mlir::Value":$input, "ArrayRef<OpFoldResult>":$offsets,
+      "ArrayRef<OpFoldResult>":$sizes, "ArrayRef<OpFoldResult>":$strides,
+      "::mlir::Value":$bd_id, "::mlir::Value":$channel)>,
+    // Build a NpuHalfDmaCpyNdOp with static entries.
+    OpBuilder<(ins "::mlir::TypeRange":$result_types, "Value":$connection,
+      "::mlir::Value":$target, "ArrayRef<int64_t>":$offsets,
+      "ArrayRef<int64_t>":$sizes, "ArrayRef<int64_t>":$strides,
+      "::mlir::Value":$bd_id, "::mlir::Value":$channel)>,
+    // Build a NpuHalfDmaCpyNdOp with dynamic entries.
+    OpBuilder<(ins "::mlir::TypeRange":$result_types, "Value":$connection,
+      "::mlir::Value":$input, "ValueRange":$offsets, "ValueRange":$sizes,
+      "ValueRange":$strides, "::mlir::Value":$bd_id, "::mlir::Value":$channel)>
+  ];
+
+  let extraClassDeclaration = [{
+    /// Return the number of leading operands before the `offsets`, `sizes` and
+    /// and `strides` operands.
+    static unsigned getOffsetSizeAndStrideStartOperandIndex() { return 2; }
```

Comment:
not used?

Reply:
This needs to be implemented for the `OffsetSizeAndStrideOpInterface`.

Reply:
Ah duh whoops